homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,225 @@
+                    functools.partial(get_mac_address, hostname=user_input[CONF_HOST])
+                )
+
+            # check exist
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+
+            # save for pairing
+            self.user_input = user_input
+
+            return await self.async_step_pairing()
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_pairing(self, user_input=None):
+        """Display pairing form."""
+        errors = {}
+
+        host = self.user_input.get(CONF_HOST)
+        _LOGGER.debug("LG webOS TV %s needs to be paired", host)
+
+        if user_input is not None:
+            try:
+                client = await async_control_connect(self.hass, host, None)
+            except PyLGTVPairException:
+                return self.async_abort(reason="error_pairing")
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+
+            if "base" not in errors:
+                return await self.async_step_register(self.user_input, client)
+
+        return self.async_show_form(
+            step_id="pairing",
+            errors=errors,
+        )
+
+    async def async_step_register(self, user_input, client=None):
+        """Register entity."""
+        if client.is_registered():
+            name = user_input.get(CONF_NAME, CONF_HOST)
+            user_input["model"] = client.software_info.get("model_name")
+            maj_v = client.software_info.get("major_ver")
+            min_v = client.software_info.get("minor_ver")
+            user_input[CONF_CLIENT_SECRET] = client.client_key
+            user_input["sw_version"] = f"{maj_v}.{min_v}"
+
+            return self.async_create_entry(title=name, data=user_input)
+
+        return await self.async_step_user()
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered Webostv device."""
+        user_input = {
+            CONF_HOST: urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname,
+            CONF_NAME: discovery_info["friendlyName"],
+        }
+        self.context["title_placeholders"] = {"name": user_input[CONF_NAME]}
+
+        return await self.async_step_user(user_input)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.options = config_entry.options
+        self.host = config_entry.data[CONF_HOST]
+        self.key = config_entry.data[CONF_CLIENT_SECRET]
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                data_input = {}
+                # Set turn on service
+                if user_input.get(TURN_ON_SERVICE):
```

Comment:
Can we change this to accept a script name instead?  Asking the user to input json into the Ui is going to be error prone.

Reply:
I replaced JSON request by entering the name of a script and I checked if it exists in Ha. But I have not found a good method to check the existence of a script except to call a hass.states.get ()
tell me if it's OK for you ?